src/Psl/Result/ResultInterface.php

```diff
@@ -55,4 +56,33 @@ public function isSucceeded(): bool;
      * @return bool - `true` if the operation failed; `false` otherwise
      */
     public function isFailed(): bool;
+
+    /**
+     * Converting a result of any type can be done by using the proceed method.
+     * The implementation will either run the `$ifSucceeded` or `$ifFailed` callback.
+     * The callback will receive the result or Exception as an argument,
+     * so that you can transform it to anything you want.
+     *
+     * @template R
+     *
+     * @psalm-param callable(T): R $ifSucceeded
+     * @psalm-param callable(Exception): R $ifFailed
+     *
+     * @psalm-return R
+     */
+    public function proceed(callable $ifSucceeded, callable $ifFailed);
```

Comment:
```suggestion
    public function proceed(callable $if_succeeded, callable $if_failed);
```

all local variables and parameters should use snake case naming.

also, don't forget that parameter names are important now that we have named parameters in PHP 8 :D 

```php

$result->proceed(ifSucceeded: $callable, ifFailed: $other); // inconsistent

$result->proceed(if_succeeded: $callable, if_failed: $other); // ok

$result->proceed(on_success: $callable, on_failure: $other); // ok
```

Reply:
Would you prefer the if* or on* version?

Reply:
whatever works for you :)